go/multipooler/server/server.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package server implements the multipooler gRPC server
+package server
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+	"path/filepath"
+	"strings"
+
+	multipoolerpb "github.com/multigres/multigres/go/pb/multipoolerservice"
+	querypb "github.com/multigres/multigres/go/pb/query"
+	"github.com/multigres/multigres/go/servenv"
+)
+
+func init() {
+	// Register the pooler service in the service map
+	servenv.InitServiceMap("grpc", "pooler")
+}
+
+// RegisterService registers the MultiPooler gRPC service with the given configuration
+func RegisterService(config *Config) {
+	servenv.OnRun(func() {
+		if servenv.GRPCServer == nil {
+			return
+		}
+
+		// Check if the pooler service should be registered
+		if servenv.GRPCCheckServiceMap("pooler") {
```

Comment:
Optional nit: return on false

Reply:
Actually, it seems like we are doing things slightly differently for different binaries. I'm going to make them uniform.